.github/workflows/npm-publish-to-npm-registry.yml

```diff
@@ -70,25 +74,38 @@ jobs:
       - name: Verify the build application
         run: cd ${{ inputs.SERVICE_LOCATION }} && npm run verify
 
-      - name: Publish package
-        run: cd ${{ inputs.SERVICE_LOCATION }} && npm publish --access public
+      - name: Publish Package
+        run: |
+          cd ${{ inputs.SERVICE_LOCATION }}
+          if [ "${{ inputs.PUBLISH_TYPE }}" = "beta" ]; then
+            echo "Publishing as beta version..."
+            npm publish --access public --tag beta
+          elif [ "${{ inputs.PUBLISH_TYPE }}" = "main" ]; then
+            echo "Publishing as main version..."
+            npm publish --access public
+          else
+            echo "Invalid PUBLISH_TYPE: ${{ inputs.PUBLISH_TYPE }}"
+            exit 1
+          fi
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
 
-      - name: Tag
+      - name: Tag Release
         run: |
-          pwd
           cd ${{ inputs.SERVICE_LOCATION }}
-          VERSION=$(grep -o '"version": "[^"]*' package.json | sed 's/"version": "//') #Replace with the versioning strategy you prefer
-          VERSION="v${VERSION}_npm"
+          VERSION=$(grep -o '"version": "[^"]*' package.json | sed 's/"version": "//')
+          if [ "${{ inputs.PUBLISH_TYPE }}" = "beta" ]; then
+            VERSION="v${VERSION}-beta"
```

Comment:
what if the version is 2nd or 3rd beta
eg:
0.9.2-beta.0
0.9.2-beta.1
0.9.2-beta.2

Reply:
@ase-101 now updated,
it will fetch the version from package.json and it will publish can you please verify again